drivers/entropy/entropy_nrf5.c

```diff
@@ -278,6 +278,12 @@ static int entropy_nrf5_get_entropy_isr(struct device *dev, uint8_t *buf, uint16
 		nrf_rng_event_clear(NRF_RNG, NRF_RNG_EVENT_VALRDY);
 		nrf_rng_task_trigger(NRF_RNG, NRF_RNG_TASK_START);
 
+		/* Clear NVIC pending bit. This ensures that a subsequent RNG event
+		 * will set the Cortex-M single-bit event register to 1 (the bit is
+		 * set when NVIC pending IRQ status is changed from 0 to 1)
+		 */
+		NVIC_ClearPendingIRQ(IRQN);
```

Comment:
Should this be before the trigger?

Reply:
Well, that's probably the "intuitive thing to do", although it, clearly, has no impact. That's because: 
- the task could be already triggered but a previous invocation - we make no assumptions the task is triggered here for the first time
- as long as it _becomes_ clear _after_ we have triggered the task and _before_ we enter the while loop containing WFE, we are certain that the event will occur

No strong opinion though

Reply:
OK